spec/reek/smells/smell_warning_spec.rb

```diff
@@ -86,6 +86,31 @@
     end
   end
 
+  context '#matches?' do
+    let(:uncommunicative) do
+      uncommunicative_name_detector = build(:smell_detector,
+                                            smell_type: 'UncommunicativeVariableName')
+      build(:smell_warning, smell_detector: uncommunicative_name_detector,
+                            message: "has the variable name '@s'",
+                            parameters: { test: 'something' })
+    end
+
+    it 'matches on class symbol' do
+      expect(uncommunicative.matches?(:UncommunicativeVariableName)).to be_truthy
+    end
+    it 'matches on class symbol and params' do
+      result = uncommunicative.matches?(:UncommunicativeVariableName, test: 'something')
+      expect(result).to be_truthy
```

Comment:
I assume you introduced the one-off variable "result" because otherwise the line would have been too long and rubocop would have complained, right?
Then let's rather do:

```Ruby
it 'matches on class symbol and params' do
uncommunicative.matches?(:UncommunicativeVariableName,
                         test: 'something').to be_truthy
end
```